openmdao/visualization/n2_viewer/src/N2UserInterface.js

```diff
@@ -35,7 +35,37 @@ class InfoPropYesNo extends InfoPropDefault {
 }
 
 // Used to format that floats displayed
-let val_formatter = d3.format("g");
+let val_float_formatter = d3.format("g");
+
+/** Stringify in the JSON sense except when val is 'nan'.
+ * @param {val} string, int,... The item to convert.
+ * @returns {str} the string of the Stringify item.
+ */
+function stringify_with_support_for_nan(val){
+    if (val === 'nan'){
+       return val;
+    } else {
+       return JSON.stringify(val);
+    }
+}
+
+/** Convert an element to a string that is human readable.
+ * @param {element} string, int,... The item to convert. Not an array!
+ * @returns {str} the string of the converted element.
+ */
+function element_to_string(element){
+    let val_string;
+    if (typeof element === 'number'){
+       if (Number.isInteger(element)) {
+          val_string = element.toString();
+       } else { /* float */
+          val_string = val_float_formatter(element);
+       }
+    } else {
+       val_string = stringify_with_support_for_nan(element);
```

Comment:
this function is only used in this one place
since it's a simple if/else, I think you could just inline it and avoid a function call